lib/foreman/util.rb

```diff
@@ -27,5 +27,17 @@ def secure_token
     def secure_encryption_key
       SecureRandom.hex(ActiveSupport::MessageEncryptor.key_len / 2)
     end
+
+    # Adds a ca cert bundle with multiple ca certs to a
+    # OpenSSL::X509::Store certificate
+    def self.add_ca_bundle_to_store(ca_bundle, cert_store)
+      file = Tempfile.open('cert.pem', Rails.root.join('tmp')) do |f|
+        f.write(ca_bundle)
+        f.flush
+        f
+      end
+      cert_store.add_file(file.path)
+      file.unlink
```

Comment:
This can leak temporary files if something raises an exception. Is there a reason you can't call `cert_store.add_file(file.path)` inside the block (after flushing)?